### PR TITLE
feat: improve `pixi search` with platform selection and making limit optional

### PR DESCRIPTION
**Pull request by wolfv**
fixes: #976 
fixes: #927

### CODE REVIEW

**Review by baszalmstra — APPROVED** [review]
Thank you! This is great!

File: src/cli/search.rs [hunk 1]
```diff
@@ -34,9 +34,13 @@ pub struct Args {
     #[arg(long)]
     pub manifest_path: Option<PathBuf>,
 
+    /// The platform to search for, defaults to current platform
+    #[arg(long, default_value_t = Platform::current())]
+    pub platform: Platform,
```
Comment (ruben-arts) [inline src/cli/search.rs:39]:
```suggestion
    #[arg(short, long, default_value_t = Platform::current())]
    pub platform: Platform,
```

File: docs/cli.md [hunk 2]
```diff
@@ -419,12 +419,13 @@ Search a package, output will list the latest version of the package.
 
 - `--manifest-path <MANIFEST_PATH>`: the path to `pixi.toml`, by default it searches for one in the parent directories.
 - `--channel <CHANNEL> (-c)`: specify a channel that the project uses. Defaults to `conda-forge`. (Allowed to be used more than once)
-- `--limit <LIMIT> (-l)`: Limit the number of search results (default: 15)
+- `--limit <LIMIT> (-l)`: Optionally limit the number of search results
```
Comment (ruben-arts) [inline docs/cli.md:422]:
```suggestion
- `--limit <LIMIT> (-l)`: optionally limit the number of search results
- `--platform <PLATFORM> (-p)`: specify a platform that you want to search for. (default: current platform) 
```

**Review by ruben-arts — CHANGES_REQUESTED** [review]
Small changes

**Review by ruben-arts — APPROVED** [review]

**Pull request merged**